### PR TITLE
chore: add semgrep

### PR DESCRIPTION
**Pull request by mr-uniswap**
## Background
Adds semgrep code scanning during the build flow

### CODE REVIEW

**Review by kristiehuang — APPROVED** [review]

File: .github/workflows/semgrep.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Semgrep
+on:
+  workflow_dispatch: {}
+  pull_request: {}
+  push:
+    branches:
+      - main
+  schedule:
+    # random HH:MM to avoid a load spike on GitHub Actions at 00:00
+    - cron: '35 11 * * *'
```
Comment (adjkant) [inline .github/workflows/semgrep.yml:10]: Wouldn't we want to run this on PR's, not on merge to main or a schedule?

Comment (mr-uniswap) [reply .github/workflows/semgrep.yml:10]: It happens on both

Comment (adjkant) [reply .github/workflows/semgrep.yml:10]: As written I think this runs on pushes to main, not on open PR's

Comment (adjkant) [reply .github/workflows/semgrep.yml:10]: Ah wait I didn't see above, i'm silly

**Review by adjkant — APPROVED** [review]

**Pull request merged**